src/lib.rs

```diff
@@ -57,10 +68,94 @@ pub struct Kaboodle {
     broadcast_port: u16,
     self_addr: Option<SocketAddr>,
     cancellation_tx: Option<Sender<()>>,
+    discovery_tx: Arc<Mutex<DiscoverySenders>>,
+    departure_tx: Arc<Mutex<DepartureSenders>>,
+    fingerprint_tx: Arc<Mutex<FingerprintSenders>>,
     interface: Interface,
     identity: Bytes,
 }
 
+/// Listens to changes on the given known_peers ObservableHashMap and turns them into higher-level
+/// channel transmissions for peer arrivals, departures, and changes to the mesh fingerprint.
+fn handle_known_peers_events(
+    known_peers: Arc<Mutex<KnownPeers>>,
+    discovery_tx: Arc<Mutex<DiscoverySenders>>,
+    departure_tx: Arc<Mutex<DepartureSenders>>,
+    fingerprint_tx: Arc<Mutex<FingerprintSenders>>,
+) {
+    /// Sends the given payload to the given list of channels.
+    fn broadcast_to_channels<T>(payload: T, channels: &mut Vec<UnboundedSender<T>>)
+    where
+        T: Clone,
+    {
+        let mut closed_channels: Vec<usize> = vec![];
+        for (idx, tx) in channels.iter().enumerate() {
+            if tx.send(payload.clone()).is_err() {
+                // Channel must be closed ¯\_(ツ)_/¯
+                closed_channels.push(idx);
```

Comment:
Right; this would happen if people drop their receiver, which seems like a hard thing to do by accident. I originally did a sweep through the list and removed receivers which were closed, but it seemed easier and cleaner to just do this.